src/main.js

```diff
@@ -0,0 +1,381 @@
+const core = require("@actions/core");
+const fs = require("fs");
+const isEqual = require("lodash/isEqual");
+const artifact = require("@actions/artifact");
+const { detailedDiff } = require("deep-object-diff");
+const artifactClient = artifact.create();
+const _ = require("lodash");
+const {
+  getAllTransformations,
+  getAllLibraries,
+  createTransformation,
+  createLibrary,
+  updateTransformation,
+  updateLibrary,
+  testTransformationAndLibrary,
+  publish,
+} = require("./apiCalls");
+
+const testOutputDir = "./test-outputs";
+const uploadTestArtifact =
+  core.getInput("uploadTestArtifact")?.toLowerCase() == "true";
+const metaFilePath = core.getInput("metaPath");
+
+const testOnly = process.env.TEST_ONLY !== "false";
+const commitId = process.env.GITHUB_SHA || "";
+
+// Load transformations and libraries from a local meta file.
+function getTransformationsAndLibrariesFromLocal(fpath) {
+  core.info(`Loading transformations and libraries locally from: ${fpath}`);
+
+  const transformations = [];
+  const libraries = [];
+
+  let meta = JSON.parse(fs.readFileSync(fpath, "utf-8"));
+  if (meta.transformations) {
+    transformations.push(...meta.transformations);
+  }
+  if (meta.libraries) {
+    libraries.push(...meta.libraries);
+  }
+  return { transformations, libraries };
+}
+
+function buildNameToIdMap(arr) {
+  return arr.reduce((map, entry) => {
+    map[entry.name] = entry.id;
+    return map;
+  }, {});
+}
+
+// Fetch transformation and libraries.
+async function loadTransformationsAndLibraries() {
+  let workspaceTransformations = [];
+  let workspaceLibraries = [];
+
+  const transformationsResponse = await getAllTransformations();
+  workspaceTransformations = transformationsResponse.data
+    ? JSON.parse(JSON.stringify(transformationsResponse.data.transformations))
+    : [];
+
+  const librariesResponse = await getAllLibraries();
+  workspaceLibraries = librariesResponse.data
+    ? JSON.parse(JSON.stringify(librariesResponse.data.libraries))
```

Comment:
parse and stringify looks redundant

Reply:
This is done previously and didn't change anything regarding this.